src/ol/source/ImageStatic.js

```diff
@@ -43,7 +43,49 @@ class Static extends ImageSource {
     const /** @type {import("../Image.js").LoadFunction} */ imageLoadFunction =
         options.imageLoadFunction !== undefined
           ? options.imageLoadFunction
-          : defaultImageLoadFunction;
+          : (image, src) => {
+              if (
+                crossOrigin !== null ||
+                location?.origin === new URL(src, location?.origin).origin
+              ) {
+                fetch(src, {
+                  mode: crossOrigin === null ? 'cors' : 'same-origin',
```

Comment:
Is this correct or even necessary?  `crossOrigin === null` is not a CORS request.  It does not fetch a real SVG https://upload.wikimedia.org/wikipedia/commons/1/18/British_National_Grid.svg

Reply:
It's indeed not necessary, and I have added tests for data uri svg, same-origin svg and cors svg.